mmvizutil/security/access.py

```
@@ -42,7 +42,9 @@ def has_permission(self):
     def dispatch(self, request, *args, **kwargs):
 
         if not self.request.session.get('user', None): #check if logged in
-            return redirect(self.get_login_url())
+            # Added to break potential redirect loops.
+            if 'rd' not in request.GET:
```

Where will the user end up going if the `rd` parameter isn't in the request? Are we able to send them to the home page?

---

When rd exists the app stays on the current page.
When it is missing it redirects to the login page.

---

I am now wondering if this is a security loop hole.

---

where is the 'rd' come? a parameter in url?

---

Yes, 'rd' would be a parameter added to the url after the first authentication attempt. If Okta redirects with a successful login, but we still aren't able to place the user in the session because the state switched from http to https, then we don't want to go back to Okta repeat the loop.

---

@kanastasakis26 - I agree this would be a security hole. We shouldn't allow it to pass through, but instead we should raise an exception letting the user know that their login attempt was unsuccessful:

```
if not self.request.session.get('user', None): #check if logged in

	if 'rd' in request.GET:
		raise Exception('Something went wrong. We were unable to successfully log you into the application.')

	return redirect(self.get_login_url())
```